demo/fake_server.ex

```diff
@@ -56,22 +56,97 @@ defmodule Wanda.Executions.FakeServer do
 
   defp complete_fake_execution(execution_id, group_id, targets) do
     result = Enum.random([:passing, :warning, :critical])
-    check_results = build(:check_results_from_targets, targets: targets, result: result)
+    check_results = check_results_from_targets(targets, result)
 
     build_result =
       build(:result,
         check_results: check_results,
         execution_id: execution_id,
-        group_id: group_id,
-        result: result
+        group_id: group_id
       )
 
     Executions.complete_execution!(
       execution_id,
-      build_result
+      Map.put(build_result, :result, get_result(build_result))
```

Comment:
My logic here was that:
```
    check_results = check_results_from_targets(targets, result)

    build_result =
      build(:result,
        check_results: check_results,
        execution_id: execution_id,
        group_id: group_id
      )
``` would get me a check_results inside the results, but that the result itself wouldn't be consistent with the content of the checks_results, because I see in the factory:
```
  def result_factory do
    %Result{
      execution_id: UUID.uuid4(),
      group_id: UUID.uuid4(),
      check_results: build_list(5, :check_result),
      result: :passing
    }
  end
  ```
  
where result is set to `:passing` regardless of the checks_results it contains. I think I'm probably not understanding the meaning of that `result: :passing`

Should I just remove the `get_result` and `result_weight` functions?

Reply:
It would be:
```
build_result =
      build(:result,
        check_results: check_results,
        execution_id: execution_id,
        group_id: group_id,
        result: result
      )
```
Simply, because the generated expectations and results are based already in the previously randomized result, so we don't need to evaluate again.

PD: yes, remove `get_result` and `result_wight`